plotters/src/style/colors/colormaps.rs

```diff
@@ -0,0 +1,252 @@
+            ((FloatType::one() - $relative_difference) * FloatType::from_u8($colors[$index_upper].2).unwrap() + $relative_difference * FloatType::from_u8($colors[$index_lower].2).unwrap()).round().to_u8().unwrap()
+        )
+    };
+    ($relative_difference:expr, $colors:expr, $index_upper:expr, $index_lower:expr, RGBAColor) => {
+        RGBAColor(
+            // These equations are a very complicated way of writing a simple linear extrapolation with lots of casting between numerical values
+            // In principle every cast should be safe which is why we choose to unwrap
+            //           (1.0  - r)                   *                                           color_value_1  +                   r *                                           color_value_2
+            ((FloatType::one() - $relative_difference) * FloatType::from_u8($colors[$index_upper].0).unwrap() + $relative_difference * FloatType::from_u8($colors[$index_lower].0).unwrap()).round().to_u8().unwrap(),
+            ((FloatType::one() - $relative_difference) * FloatType::from_u8($colors[$index_upper].1).unwrap() + $relative_difference * FloatType::from_u8($colors[$index_lower].1).unwrap()).round().to_u8().unwrap(),
+            ((FloatType::one() - $relative_difference) * FloatType::from_u8($colors[$index_upper].2).unwrap() + $relative_difference * FloatType::from_u8($colors[$index_lower].2).unwrap()).round().to_u8().unwrap(),
+            ((FloatType::one() - $relative_difference) * FloatType::from_f64($colors[$index_upper].3).unwrap() + $relative_difference * FloatType::from_f64($colors[$index_lower].3).unwrap()).to_f64().unwrap()
+        )
+    };
+    ($relative_difference:expr, $colors:expr, $index_upper:expr, $index_lower:expr, HSLColor) => {
+        HSLColor(
+            // These equations are a very complicated way of writing a simple linear extrapolation with lots of casting between numerical values
+            // In principle every cast should be safe which is why we choose to unwrap
+            //           (1.0  - r)                   *                                           color_value_1  +                   r *                                           color_value_2
+            ((FloatType::one() - $relative_difference) * FloatType::from_f64($colors[$index_upper].0).unwrap() + $relative_difference * FloatType::from_f64($colors[$index_lower].0).unwrap()).to_f64().unwrap(),
+            ((FloatType::one() - $relative_difference) * FloatType::from_f64($colors[$index_upper].1).unwrap() + $relative_difference * FloatType::from_f64($colors[$index_lower].1).unwrap()).to_f64().unwrap(),
+            ((FloatType::one() - $relative_difference) * FloatType::from_f64($colors[$index_upper].2).unwrap() + $relative_difference * FloatType::from_f64($colors[$index_lower].2).unwrap()).to_f64().unwrap(),
+        )
+    };
+);
+
+
+macro_rules! implement_color_scale_for_derived_ColorMap{
+    ($($color_type:tt),+) => {
```

Comment:
I have corrected it :+1: It does not appear to be working with `ty` but rather with `ident`. This is probably due to how I match the supplied patterns. eg these lines:
```
($relative_difference:expr, $colors:expr, $index_upper:expr, $index_lower:expr, HSLColor) => {
```
I could not manage to make it work by actually accepting a type instead of an identifier but am not sure why.